controllers/submission.go

```diff
@@ -71,3 +71,56 @@ func PostSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request)
 		return http.StatusBadRequest, errors.New("Unknown submission kind.")
 	}
 }
+
+// FinalSubmission makes the last submission final.
+func FinalSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request) (status int, err error) {
+	p, ok := passenger.FromContext(ctx)
+	if !ok {
+		return http.StatusUnauthorized, nil
+	}
+	if r.Method != "POST" {
+		return http.StatusMethodNotAllowed, nil
+	}
+
+	resultKey, err := datastore.DecodeKey(mux.Vars(r)["resultKey"])
+
+	if !util.HasParent(p.UserKey, resultKey) {
+		return http.StatusBadRequest, errors.New("Cannot submit answer for other users")
```

Comment:
[Error strings should not be capitalized](https://github.com/golang/lint/blob/master/lint.go#L1163).